[iOS 16] All layout tests in fast/events/ios/rotation time out

whsieh opened the pull request:
#### b751dd03ccdb9abb0eefb3baa75d3cfe0ddc877a
<pre>
[iOS 16] All layout tests in fast/events/ios/rotation time out
<a href="https://bugs.webkit.org/show_bug.cgi?id=242199">https://bugs.webkit.org/show_bug.cgi?id=242199</a>
rdar://93784344

Reviewed by Devin Rousso and Tim Horton.

For all apps linked on or after iOS 16, `-[UIDevice setOrientation:animated:]` is a no-op; this
causes all layout tests that attempt to simulate rotation (i.e., all the tests inside
fast/events/ios/rotation) to time out while waiting for the web view to resize during rotation.

In its place, UIKit has added `-[UIWindowScene requestGeometryUpdateWithPreferences:errorHandler:]`,
which can be used to request a scene-level transition to one or more requested interface
orientations; fix these failing layout tests by adopting this API instead on iOS 16.

* Source/WTF/wtf/PlatformHave.h:

Add a new build-time flag to guard the availability of the new UIKit API.

* Tools/WebKitTestRunner/ios/TestControllerIOS.mm:
(WTR::restorePortraitOrientationIfNeeded):
(WTR::TestController::platformResetStateToConsistentValues):

Additionally refactor logic for restoring device orientation to portrait mode between tests, so that
it&apos;s also compatible with iOS 16.

* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::simulateRotation):
(WTR::UIScriptControllerIOS::simulateRotationLikeSafari):

Factor out common logic into a private helper method; use `-requestGeometryUpdateWithPreferences:`
on iOS 16, and fall back to `-setOrientation:animated:` elsewhere.

(WTR::toWindowSceneGeometryPreferences):
(WTR::toUIDeviceOrientation):

Canonical link: <a href="https://commits.webkit.org/252016@main">https://commits.webkit.org/252016@main</a>
</pre>

dcrousso commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1256,9 +1256,14 @@
 
-#if !defined(HAVE_UIKIT_RESIZABLE_WINDOWS) && PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000
+#if PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000
+#if !defined(HAVE_UIKIT_RESIZABLE_WINDOWS)
```
NIT: Is checking `!defined` really necessary here?  Most other things in this file don't seem to bother 🤔

whsieh replied:
I don't see anything in the style guidelines (or webkit-dev) about this 🤷🏻‍♂️

whsieh committed:
[iOS 16] All layout tests in fast/events/ios/rotation time out
https://bugs.webkit.org/show_bug.cgi?id=242199
rdar://93784344

Reviewed by Devin Rousso and Tim Horton.

For all apps linked on or after iOS 16, `-[UIDevice setOrientation:animated:]` is a no-op; this
causes all layout tests that attempt to simulate rotation (i.e., all the tests inside
fast/events/ios/rotation) to time out while waiting for the web view to resize during rotation.

In its place, UIKit has added `-[UIWindowScene requestGeometryUpdateWithPreferences:errorHandler:]`,
which can be used to request a scene-level transition to one or more requested interface
orientations; fix these failing layout tests by adopting this API instead on iOS 16.

* Source/WTF/wtf/PlatformHave.h:

Add a new build-time flag to guard the availability of the new UIKit API.

* Tools/WebKitTestRunner/ios/TestControllerIOS.mm:
(WTR::restorePortraitOrientationIfNeeded):
(WTR::TestController::platformResetStateToConsistentValues):

Additionally refactor logic for restoring device orientation to portrait mode between tests, so that
it's also compatible with iOS 16.

* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.h:
* Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm:
(WTR::UIScriptControllerIOS::simulateRotation):
(WTR::UIScriptControllerIOS::simulateRotationLikeSafari):

Factor out common logic into a private helper method; use `-requestGeometryUpdateWithPreferences:`
on iOS 16, and fall back to `-setOrientation:animated:` elsewhere.

(WTR::toWindowSceneGeometryPreferences):
(WTR::toUIDeviceOrientation):

Canonical link: https://commits.webkit.org/252016@main
